packages/slate-commons/src/commands/replaceNode.tsx

```diff
@@ -0,0 +1,27 @@
+import { Editor, Transforms } from 'slate';
+import type { NodeMatch, Node, NodeEntry } from 'slate';
+
+interface Props<T extends Node> {
+    entry: NodeEntry<T>;
+    match: NodeMatch<T>;
+}
+
+export function replaceNode<Original extends Node, New extends Node>(
+    editor: Editor,
+    props: Props<Original>,
+    createNode: () => New,
```

Comment:
I guess it would make more sense to pass the actual new node here (not a factory).

```ts
export function replaceNode<Original extends Node, New extends Node>(
    editor: Editor,
    props: Props<Original>,
    node: Omit<New, 'children'>,
)